### ISSUE TITLE
Feature request: return the number of failed attempts

### ISSUE THREAD

**Issue by louisabraham** [anthropics/anthropic-sdk-python#608]
You allow a parameter "max_retries". It would be great to return along with the API answer the number of "failed" API calls so that people have an idea of the load.

I don't know what your policy towards outside contributions is but if you agree I can implement it.

**Comment by RobertCraigie** [comment]
Hey @louisabraham, that's a good idea! I think we'd want to add this to our [raw response](https://github.com/anthropics/anthropic-sdk-python#accessing-raw-response-data-eg-headers) API, e.g. 
```py
response = client.messages.with_raw_response.create(...)
response.retry_count  # 0
```
Thanks for offering to work on a PR! That would be very helpful :)

I'll discuss this with a couple people on our end to land on the exact naming we want but feel free to work on this in the meantime and we'll handle any renaming :)

**Comment by louisabraham** [comment]
Thanks! It looks like `with_raw_response` is supposed to be deprecated, is this a good idea to implement new features in it?

**Comment by RobertCraigie** [comment]
Certain functions / behaviour with it is deprecated as we want to have separate classes for the sync & async client in the future. What gave you the impression that `with_raw_response` as a whole is deprecated? Definitely don't want our docs to convey that 😅

**Comment by louisabraham** [comment]
Thank you for your answer!

> These methods return an LegacyAPIResponse object. This is a legacy class as we're changing it slightly in the next major version.

I think the term "legacy" misled me!

**Comment by louisabraham** [comment]
I just assigned @NDV-tiime, he will look at this issue next week.

**Comment by RobertCraigie** [comment]
This was released in `v0.33.0`!

**Comment by louisabraham** [comment]
Thank you all!
